reader_concurrency_semaphore.cc

```diff
@@ -1507,6 +1514,12 @@ void reader_concurrency_semaphore::maybe_admit_waiters() noexcept {
         auto& permit = _wait_list.front();
         dequeue_permit(permit);
         try {
+            // Abort the read if its remaining time is less than half of its timeout when arrived to the semaphore
+            if (permit.timeout() - db::timeout_clock::now() <= (permit.timeout() - permit.created())/2) {
```

Comment:
Let's make that 0.5 a parameter. I can think about workloads that have tiny timeouts that will regress due to this, so better have a way to back out.